typst: new, 0.0.0+git2023032

Nyovelt opened the pull request:
<!-- For description on topic creation and maintenance, please refer to [this Wiki article](https://wiki.aosc.io/developer/packaging/topic-based-maintenance-guideline/). -->

Trying to introduce typst, A LaTeX killer.

Topic Description
-----------------

<!-- Please input topic description here. -->
No
Package(s) Affected
-------------------

<!-- Please list all package(s) affected by this topic here. -->
No
Security Update?
----------------

<!-- If this topic contains security update(s), please uncomment "Yes,"
     mark with the `security` and `priority` labels, and make sure to mark your commits to relevant issue numbers.

     Please see GitHub's documentation on "Linking a pull request to an issue":

     https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue -->

<!-- Yes - Issue Number: ISSUENUMBER -->
<!-- No -->

<!-- Please uncomment the "Build Order" section if your topic affects more than one package. -->

<!--
Build Order
-----------

Please describe in what order maintainers should build this pull request.
-->

Test Build(s) Done
------------------

**Primary Architectures**

<!-- Please remove any architecture to which this topic does not apply. -->

- [x] AMD64 `amd64`   
- [ ] AArch64 `arm64`
<!-- If this package involves a `+32` counterpart, please uncomment the line below. -->    
<!-- - [ ] 32-bit Optional Environment `optenv32` -->

<!-- If all package(s) affected by this topic is `noarch`, please use the stub below. -->
<!-- - [ ] Architecture-independent `noarch` -->

**Secondary Architectures**

<!-- Please remove any architecture to which this topic does not apply. -->

Architectural progress for "secondary," or experimental ports does not impede on merging of this topic.

- [ ] Loongson 3 `loongson3`
- [ ] MIPS R6 64-bit (Little Endian) `mips64r6el`
- [ ] PowerPC 64-bit (Little Endian) `ppc64el`
- [ ] RISC-V 64-bit `riscv64`

<!-- Maintainers should review file changes and, if the build script(s) affected complies with the
     [Package Styling Manual](https://wiki.aosc.io/developer/packaging/package-styling-manual/), please
     add `lgtm` label to this issue. -->

<!-- Maintainers and users may now test the packages in this topic and, once user/maintainer feedback indicates that the update(s) work as expected and find its quality satisfactory,
     another maintainer may now review this pull request and mark it as Approved. After which, the maintainer will build affected package(s) and upload them to the `stable` repository. -->

Update(s) Uploaded to Stable
----------------------------

**Primary Architectures**

<!-- Please remove any architecture to which this topic does not apply. -->

- [ ] AMD64 `amd64`   
- [ ] AArch64 `arm64`
<!-- If this package involves a `+32` counterpart, please uncomment the line below. -->
<!-- - [ ] 32-bit Optional Environment `optenv32` -->

<!-- If all package(s) affected by this topic is `noarch`, please use the stub below. -->
<!-- - [ ] Architecture-independent `noarch` -->

**Secondary Architectures**

<!-- Please remove any architecture to which this topic does not apply. -->

Architectural progress for "secondary," or experimental ports does not impede on merging of this topic.

- [ ] Loongson 3 `loongson3`
- [ ] MIPS R6 64-bit (Little Endian) `mips64r6el`
- [ ] PowerPC 64-bit (Little Endian) `ppc64el`
- [ ] RISC-V 64-bit `riscv64`

<!-- TODO: CI to auto-fill architectural progress. -->

Nyovelt commented:
![](https://webresources.aaaab3n.moe/2023/03/b8e0443386d5bdc934c7b24b7f05fd25.png)

Nyovelt commented:
![](https://webresources.aaaab3n.moe/2023/03/0eee352b6892caf1518f543f2e8d0017.png)

MingcongBai commented on app-doc/typst/spec:
```diff
@@ -0,0 +1,4 @@
+VER=0.0.0+git2023032
```
Upstream does have a tag - https://github.com/typst/typst/tags.

Nyovelt replied:
The thing is, based on this commit, https://github.com/typst/typst/commits/main, the release cannot be compiled unless patched to the latest commit.

MingcongBai commented on app-doc/typst/spec:
```diff
@@ -0,0 +1,4 @@
+VER=0.0.0+git2023032
+REL=1
+SRCS="git::commit=a253b47e7c6f5e9500a40ac73f949a431f62289d::https://github.com/typst/typst.git"
+CHKSUMS="SKIP"
```
CHKUPATE missing.

MingcongBai commented on app-doc/typst/autobuild/defines:
```diff
@@ -0,0 +1,8 @@
+PKGNAME=typst
+PKGSEC=doc
+PKGDES="A new markup-based typsetting system that is designed to be as powerful as LaTeX while being much easier to learn and use."
```
This is too verbose for a short description and it forces a lot of subjective descriptions. Try this...

A markup-based typsetting system

This should be good enough.

Nyovelt committed:
typst: new, 23.03.21.2+git20230327

According to https://github.com/typst/typst/pull/173, the release cannot be compiled unless the code is patched to the latest commit

TODO: When the issue was fixed in a new release, the SRC file should be updated to reference the release file rather than the Git snapshot.
